### DISCUSSION TITLE
[No driver] No xpath selectors feature, only css selectors?

### DISCUSSION THREAD

**Discussion by Samstix636** [ultrafunkamsterdam/undetected-chromedriver#2016] [Q&A]
Hello guys, been going over the nodriver documentation and it looks like the tab.select() function only works with css selectors based on the documentation? 
Is there a way to find elements using xpath selectors? Would really appreciate any help

**Comment by devblack** [comment]
You can use https://ultrafunkamsterdam.github.io/nodriver/nodriver/cdp/dom.html#nodriver.cdp.dom.perform_search

**Comment by arthurauffray** [comment]
@ultrafunkamsterdam Isn't xpath the best way to locate elements?

**Comment by ultrafunkamsterdam** [comment]
You can just use .find() and .find_all()  and use xpath instead of search keyword

**Reply by ultrafunkamsterdam** [reply]
eg: ```await tab.find_all('//*')```
---
to use more advanced xpaths you will first make the call
```await tab.send(cdp.dom.enable())```


then you can do stuff like ```await tab.find_all('//div[text()]')```  
or even go completely overboard:
```
await tab.find_all('/html/body//text()[ contains( translate(., "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "abcdefghijklmnopqrstuvwxyz"), "test") ]' )
```
```
Out[21]:
[<a class="ico-webrtc" href="/webrtc" title="WebRTC Leak Test">WebRTC Leak Test</a>,
 <a class="ico-tls" href="/tls" title="SSL/TLS Client Security Test">SSL/TLS Client Test</a>]
 ```

**Reply by ultrafunkamsterdam** [reply]
@arthurauffray if it's the best way...  i personally don't think so. It's too steep of a learning curve and error prone.

I prefer good old css selectors. If can't locate an element, i locate one in close proximity and use eg elem.parent.parent  and elem.children[0].children[-1] .  
or of course, goold old text search
